src/app/users/layout.tsx

```diff
@@ -0,0 +1,25 @@
+'use client';
+
+import { PropsWithChildren, useEffect } from 'react';
+
+import { useParams, useRouter } from 'next/navigation';
+
+import { useGetUserStatusAPI } from '@/src/apis/myInfo';
+
+const UsersLayout = ({ children }: PropsWithChildren) => {
+  const { userId } = useParams() as { userId: string };
+  const router = useRouter();
+  const { login: isLogin, userId: currentUserId } = useGetUserStatusAPI();
+
+  useEffect(() => {
+    if (isLogin) {
+      if (userId === String(currentUserId)) {
+        router.replace('/users/profile');
+      }
+    }
+  }, [isLogin, currentUserId, router, userId]);
```

Comment:
```suggestion
if (isLogin && userId === String(currentUserId)) router.replace('/users/profile');
```

useEffect 를 제거할수도 있을것 같습니다!

Reply:
오 그러네요 알겠습니다